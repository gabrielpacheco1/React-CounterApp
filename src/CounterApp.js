import React, {useState} from 'react'
import PropTypes from 'prop-types'

//SNIPPET: rafcp (crea un funct comp con proptypes)

const CounterApp = ({value=0}) => {


    /*const state = useState('Gabriel'); //usualmente no se usa asi
    console.log(state)*/

    /*const [nombre, setNombre] = useState('Gabriel');
    console.log(nombre, setNombre)*/

    const [counter, setCounter] = useState(value);

    const handleAdd = (e) => {
        setCounter(counter+1);
        //setCounter( (c) => c+1 ); otra forma
    } 

    const handleSubstract = (e) => {
        setCounter(counter-1);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h1 className="counter">{counter}</h1>

            <button className= "contar" onClick= {handleSubstract}>-1</button>
            <button onClick= { () => setCounter(value)}>Reset</button>
            <button className= "contar" onClick= {handleAdd}>+1</button>
            
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number
}

export default CounterApp;