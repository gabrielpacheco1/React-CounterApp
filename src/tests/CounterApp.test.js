import React from 'react';
//import {render} from '@testing-library/react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import CounterApp from '../CounterApp'

describe('Pruebas en comp CounterApp', () => {
    
    let wrapper= shallow(<CounterApp />);

    beforeEach(() => {
        wrapper= shallow(<CounterApp />);
    })

    test('debe mostar el <CounterApp /> correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostar el valor por defecto de 100', () => {
        const valorDefecto= 100;
        const wrapper= shallow(<CounterApp value= {valorDefecto} />)
    
        const valorContador= wrapper.find('h1.counter').text(); //similar al querySelector
        //console.log(valorContador);
        expect(valorContador).toBe(valorDefecto.toString());
    })
    
    test('debe incrementar con el boton +1', () => {
        wrapper.find('button').at(2).simulate('click')
        //console.log(buttonMas.html()); //para comprobar que estoy recuperandolo correctamente
        const valorContador= wrapper.find('h1.counter').text().trim();
        expect(valorContador).toBe('1');
    })
    
    test('debe decrementar con el boton -1', () => {
        wrapper.find('button').at(0).simulate('click')
        const valorContador= wrapper.find('h1.counter').text().trim();
        expect(valorContador).toBe('-1');
    })

    test('debe colocar el valor por def con el boton reset', () => {
        const wrapper= shallow(<CounterApp value= {150} />)
        wrapper.find('button').at(0).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        const valorContador= wrapper.find('h1.counter').text().trim();
        console.log(valorContador);

    })
    
})

