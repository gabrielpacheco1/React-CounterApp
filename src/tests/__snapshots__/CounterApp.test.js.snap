// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en comp CounterApp debe mostar el <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h1
    className="counter"
  >
    0
  </h1>
  <button
    className="contar"
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    className="contar"
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
